sdk/translation/Azure.AI.Translation.Document/tests/GetAllTranslationsFilterTests.cs

```diff
@@ -38,7 +38,29 @@ public async Task<List<string>> CreateTranslationJobsAsync(DocumentTranslationCl
                 else if (jobTerminalStatus == DocumentTranslationStatus.Canceled)
                 {
                     await translationOp.CancelAsync(default);
-                    Thread.Sleep(6000); // wait for cancel status to propagate!
+                }
+            }
+
+            //ensure that cancel status has propagated before returning
+            if (jobTerminalStatus == DocumentTranslationStatus.Canceled)
+            {
+                var options = new GetTranslationStatusesOptions();
+
+                foreach (var result in resultIds)
+                {
+                    options.Ids.Add(result);
+                }
+
+                bool cancellationHasPropagated = false; // flag for successful cancelling
+                int watchdogCounter = 0;                // counter to return in case of long term service failure
+
+                while (!cancellationHasPropagated && watchdogCounter < 100)
+                {
+                    Thread.Sleep(5000); // wait for cancel status to propagate!
```

Comment:
This is not ideal as it makes playback take longer even though the wait is not required there.

Reply:
Consider using `await TestRetryHelper.RetryAsync` https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Certificates/tests/CertificatesTestBase.cs#L227-L237

Reply:
Thank you so much for your input. It makes perfect sense. I have implemented something similar to KeyVault's `WaitForPurgedCertificate()` and created a PR for it at #23750 . Please review it if you get the chance!